explorer/src/Pos/Explorer/TestUtil.hs

```diff
@@ -0,0 +1,240 @@
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Pos.Explorer.TestUtil where
+
+import qualified Prelude
+import           Universum
+
+import           Data.Default                      (def)
+import qualified Data.List.NonEmpty                as NE
+import           Data.Text.Buildable               (build)
+import           Serokell.Data.Memory.Units        (Byte, Gigabyte, convertUnit)
+import           Test.QuickCheck                   (Arbitrary (..), Property, Testable,
+                                                    counterexample, forAll, generate,
+                                                    property)
+import           Test.QuickCheck.Arbitrary.Generic (genericArbitrary)
+
+import           Pos.Arbitrary.Block               ()
+import           Pos.Block.Core                    (Block, BlockHeader, GenesisBlock,
+                                                    MainBlock, getBlockHeader)
+import           Pos.Block.Core.Genesis.Misc       (mkGenesisBlock)
+import           Pos.Block.Logic                   (RawPayload (..), createMainBlockPure)
+import           Pos.Block.Types                   (SlogUndo, Undo)
+import qualified Pos.Communication                 ()
+import           Pos.Core                          (BlockCount (..), ChainDifficulty (..),
+                                                    EpochIndex (..), HasConfiguration,
+                                                    LocalSlotIndex (..), SlotId (..),
+                                                    SlotLeaders, StakeholderId,
+                                                    difficultyL)
+import           Pos.Crypto                        (SecretKey)
+import           Pos.Delegation                    (DlgPayload, DlgUndo, ProxySKBlockInfo)
+import           Pos.Ssc.Core                      (SscPayload, defaultSscPayload)
+import           Pos.Txp.Core                      (TxAux)
+import           Pos.Update.Configuration          (HasUpdateConfiguration)
+import           Pos.Update.Core                   (UpdatePayload (..))
+
+
+----------------------------------------------------------------
+-- Arbitrary and Show instances
+----------------------------------------------------------------
+
+-- I used the build function since I suspect that it's safe (even in tests).
+instance HasConfiguration => Prelude.Show SlogUndo where
+    show = show . build
+
+instance Prelude.Show DlgUndo where
+    show = show . build
+
+instance HasConfiguration => Prelude.Show Undo where
+    show = show . build
+
+instance Arbitrary SlogUndo where
+    arbitrary = genericArbitrary
+
+instance HasConfiguration => Arbitrary DlgUndo where
+    arbitrary = genericArbitrary
+
+instance HasConfiguration => Arbitrary Undo where
+    arbitrary = genericArbitrary
+
+----------------------------------------------------------------
+-- Utility
+-- TODO(ks): Extract this in some common PureBlockTest module in src/?
```

Comment:
It would be better to extract these function into `explorer/test/...` (not `explorer/src`), isn't? As I can see `basicBlockGenericUnsafe`, `basicBlock` etc. are used in tests only.

Reply:
Do have in mind some of these functions will probably be used in `bench/` which doesn't have access to `test`.
The idea behind the comment is to move this module somewhere globally, on the project level (like `lib/`), so we can reuse it from any module (and add other useful functions from any module).